src/main/java/wiresegal/wob/arcanum/Parsing.kt

```diff
@@ -167,6 +168,57 @@ fun about(message: Message) {
     })
 }
 
+val progressCache = mutableListOf<Pair<String, String>>()
+var progressCacheTimeStamp: Long? = null
+
+fun showProgressBar(message: Message) {
+    val projects = getProgressBar()
+    var maxLength = 0
+    for (project in projects) {
+        maxLength = Integer.max(maxLength, project.first.length)
+    }
+    var progresses = ""
+    for (project in projects) {
```

Comment:
It looks like you can combine the two `for` loops into one if the TODO below is irrelevant.